get_unpacked_marks() got an unexpected keyword argument 'consider_mro'

I just got the following in CI

```
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/pytest_asyncio/plugin.py", line 552, in pytest_collectstart
INTERNALERROR>     marks = get_unpacked_marks(collector.obj, consider_mro=True)
INTERNALERROR> TypeError: get_unpacked_marks() got an unexpected keyword argument 'consider_mro'
```

I can/will pin, but just warning you in case it matters. I am pinned to `Pytest<7.1` for various reasons, so maybe you need to have a min version of pytest ?

I have started getting a similar error in CI. The only dependency that changed since yesterday was pytest-asyncio
<details><summary>Log</summary>
<p>

```
INTERNALERROR> def worker_internal_error(self, node, formatted_error):
INTERNALERROR>         """
INTERNALERROR>         pytest_internalerror() was called on the worker.
INTERNALERROR>     
INTERNALERROR>         pytest_internalerror() arguments are an excinfo and an excrepr, which can't
INTERNALERROR>         be serialized, so we go with a poor man's solution of raising an exception
INTERNALERROR>         here ourselves using the formatted message.
INTERNALERROR>         """
INTERNALERROR>         self._active_nodes.remove(node)
INTERNALERROR>         try:
INTERNALERROR> >           assert False, formatted_error
INTERNALERROR> E           AssertionError: Traceback (most recent call last):
INTERNALERROR> E               File "/home/runner/work/cylc-flow/cylc-flow/cylc/flow/parsec/config.py", line 145, in get
INTERNALERROR> E                 cfg = cfg[key]
INTERNALERROR> E               File "/home/runner/work/cylc-flow/cylc-flow/cylc/flow/parsec/OrderedDict.py", line 38, in __getitem__
INTERNALERROR> E                 return OrderedDict.__getitem__(self, key)
INTERNALERROR> E             KeyError: '_remote_background_shared_tcp'
INTERNALERROR> E             
INTERNALERROR> E             During handling of the above exception, another exception occurred:
INTERNALERROR> E             
INTERNALERROR> E             Traceback (most recent call last):
INTERNALERROR> E               File "/home/runner/work/cylc-flow/cylc-flow/tests/unit/test_host_select_remote.py", line 48, in <module>
INTERNALERROR> E                 remote_platform = glbl_cfg().get(
INTERNALERROR> E               File "/home/runner/work/cylc-flow/cylc-flow/cylc/flow/parsec/config.py", line 151, in get
INTERNALERROR> E                 raise ItemNotFoundError(itemstr(parents, key))
INTERNALERROR> E             cylc.flow.parsec.exceptions.ItemNotFoundError: You have not set "[platforms]_remote_background_shared_tcp" in this config.
INTERNALERROR> E             
INTERNALERROR> E             During handling of the above exception, another exception occurred:
INTERNALERROR> E             
INTERNALERROR> E             Traceback (most recent call last):
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/main.py", line 271, in wrap_session
INTERNALERROR> E                 session.exitstatus = doit(config, session) or 0
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/main.py", line 324, in _main
INTERNALERROR> E                 config.hook.pytest_collection(session=session)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
INTERNALERROR> E                 return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
INTERNALERROR> E                 return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 152, in _multicall
INTERNALERROR> E                 return outcome.get_result()
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pluggy/_result.py", line 114, in get_result
INTERNALERROR> E                 raise exc.with_traceback(exc.__traceback__)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
INTERNALERROR> E                 res = hook_impl.function(*args)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/main.py", line 335, in pytest_collection
INTERNALERROR> E                 session.perform_collect()
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/main.py", line 675, in perform_collect
INTERNALERROR> E                 self.items.extend(self.genitems(node))
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/main.py", line 845, in genitems
INTERNALERROR> E                 yield from self.genitems(subnode)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/main.py", line 842, in genitems
INTERNALERROR> E                 rep = collect_one_node(node)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/runner.py", line 546, in collect_one_node
INTERNALERROR> E                 ihook.pytest_collectstart(collector=collector)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
INTERNALERROR> E                 return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
INTERNALERROR> E                 return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 113, in _multicall
INTERNALERROR> E                 raise exception.with_traceback(exception.__traceback__)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
INTERNALERROR> E                 res = hook_impl.function(*args)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pytest_asyncio/plugin.py", line 552, in pytest_collectstart
INTERNALERROR> E                 marks = get_unpacked_marks(collector.obj, consider_mro=True)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/python.py", line 310, in obj
INTERNALERROR> E                 self._obj = obj = self._getobj()
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/python.py", line 528, in _getobj
INTERNALERROR> E                 return self._importtestmodule()
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/python.py", line 617, in _importtestmodule
INTERNALERROR> E                 mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/pathlib.py", line 567, in import_path
INTERNALERROR> E                 importlib.import_module(module_name)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/importlib/__init__.py", line 127, in import_module
INTERNALERROR> E                 return _bootstrap._gcd_import(name[level:], package, level)
INTERNALERROR> E               File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
INTERNALERROR> E               File "<frozen importlib._bootstrap>", line 991, in _find_and_load
INTERNALERROR> E               File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
INTERNALERROR> E               File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/assertion/rewrite.py", line 186, in exec_module
INTERNALERROR> E                 exec(co, module.__dict__)
INTERNALERROR> E               File "/home/runner/work/cylc-flow/cylc-flow/tests/unit/test_host_select_remote.py", line 61, in <module>
INTERNALERROR> E                 pytest.skip('Remote test host not available', allow_module_level=True)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/outcomes.py", line 179, in skip
INTERNALERROR> E                 raise Skipped(msg=reason, allow_module_level=allow_module_level)
INTERNALERROR> E             Skipped: Remote test host not available
INTERNALERROR> E           assert False
INTERNALERROR> 
INTERNALERROR> /opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/xdist/dsession.py:197: AssertionError
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/main.py", line 271, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/main.py", line 325, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 130, in _multicall
INTERNALERROR>     teardown[0].send(outcome)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pluggy/_result.py", line 114, in get_result
INTERNALERROR>     raise exc.with_traceback(exc.__traceback__)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/xdist/dsession.py", line 1[22](https://github.com/cylc/cylc-flow/actions/runs/6704426380/job/18216810060#step:11:23), in pytest_runtestloop
INTERNALERROR>     self.loop_once()
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/xdist/dsession.py", line 1[45](https://github.com/cylc/cylc-flow/actions/runs/6704426380/job/18216810060#step:11:46), in loop_once
INTERNALERROR>     call(**kwargs)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.8.18/x[64](https://github.com/cylc/cylc-flow/actions/runs/6704426380/job/18216810060#step:11:65)/lib/python3.8/site-packages/xdist/dsession.py", line 1[85](https://github.com/cylc/cylc-flow/actions/runs/6704426380/job/18216810060#step:11:86), in worker_workerfinished
INTERNALERROR>     self._active_nodes.remove(node)
INTERNALERROR> KeyError: <WorkerController gw2>
```

</p>
</details>

--- a/dependencies/pytest-min/constraints.txt
+++ b/dependencies/pytest-min/constraints.txt
@@ -18,5 +18,6 @@ pytest==7.0.0
 requests==2.28.1
 sortedcontainers==2.4.0
 tomli==2.0.1
+trio==0.21.0
 urllib3==1.26.12
 xmlschema==2.1.1

--- a/docs/source/reference/changelog.rst
+++ b/docs/source/reference/changelog.rst
@@ -2,6 +2,10 @@
 Changelog
 =========
 
+0.22.1 (UNRELEASED)
+===================
+- Fixes a bug that broke compatibility with pytest>=7.0,<7.2. `#654 <https://github.com/pytest-dev/pytest-asyncio/pull/654>`_
+
 0.22.0 (2023-10-31)
 ===================
 - Class-scoped and module-scoped event loops can be requested

--- a/pytest_asyncio/plugin.py
+++ b/pytest_asyncio/plugin.py
@@ -549,7 +549,7 @@ def pytest_collectstart(collector: pytest.Collector):
         return
     # pytest.Collector.own_markers is empty at this point,
     # so we rely on _pytest.mark.structures.get_unpacked_marks
-    marks = get_unpacked_marks(collector.obj, consider_mro=True)
+    marks = get_unpacked_marks(collector.obj)
     for mark in marks:
         if not mark.name == "asyncio_event_loop":
             continue

--- a/tox.ini
+++ b/tox.ini
@@ -7,18 +7,20 @@ passenv =
 
 [testenv]
 extras = testing
-deps =
-     --requirement dependencies/default/requirements.txt
-     --constraint dependencies/default/constraints.txt
+install_command = python -m pip install \
+    --requirement dependencies/default/requirements.txt \
+    --constraint dependencies/default/constraints.txt \
+    {opts} {packages}
 commands = make test
 allowlist_externals =
     make
 
 [testenv:pytest-min]
 extras = testing
-deps =
-     --requirement dependencies/pytest-min/requirements.txt
-     --constraint dependencies/pytest-min/constraints.txt
+install_command = python -m pip install \
+    --requirement dependencies/pytest-min/requirements.txt \
+    --constraint dependencies/pytest-min/constraints.txt \
+    {opts} {packages}
 commands = make test
 allowlist_externals =
     make

